feat: Add CSharp SDK

vgwizardx opened the pull request:
[FLI-783 - C# Server Side SDK](https://github.com/flipt-io/flipt-server-sdks/issues/86)
- Add Authentication
- Add Evaluation (Boolean, Variant, and Batch)

markphelps commented on flipt-csharp/src/Clients/Evaluation.cs:
```diff
@@ -0,0 +1,155 @@
+
+            if (_authenticationStrategy != null)
+                httpRequest.Headers.Authorization =
+                    new AuthenticationHeaderValue("Bearer", _authenticationStrategy.GetAuthorizationHeader());
```
I dont think we want `Bearer` here always because depending on the authorization strategy, the header should be:

`Authorization: Bearer ${token}` for client token strategy : https://docs.flipt.io/authentication/using-tokens

or 

`Authorizatoin: JWT ${token}` for JWT strategy: https://docs.flipt.io/authentication/using-jwts

I know the JWT prefix isn't standard, but we wanted our server to easily know which type of token it was for decoding the auth instead of first trying static token then JWT as we would have to do if we used the `Bearer` prefix for both

vgwizardx replied:
fixed

markphelps commented on flipt-csharp/srcWrong/Controllers/EvaluationController.cs:
```diff
@@ -0,0 +1,75 @@
+﻿using System.Text.Json;
```
question, what is the `srcWrong` package for? im not very well versed in C# idioms so just wondering

vgwizardx replied:
Sorry didn't realize that was committed. That wasn't supposed to be apart of the commit. It was me moving and renaming stuff. I will take that folder out.

markphelps reviewed (changes requested):
looking great! thank you @vgwizardx !!

One minor change around auth headers requested, as well as a question about the `srcWrong` directory/package

Also, is there a best practice for testing in C#, like a test runner/framework/pattern? I can add some integration tests once we get this merged so just wondering.

Thanks again!

vgwizardx commented:
@markphelps I fixed the issues you mentioned. Since the srcWrong didn't show in Visual Studio, I forgot I removed it, but it must not have been deleted from the folder. Now, the header will be set by the strategy. As far as patterns for testing. Xunit is commonly used, similar to junit. 

This might be more than what you want, but you can do something like this for the integration test 
[https://blog.devgenius.io/integration-testing-in-net-8-b8ab83e7531d](https://blog.devgenius.io/integration-testing-in-net-8-b8ab83e7531d)

you can also use other integration testing tools like Selenium but I haven't set anything like that up.

markphelps reviewed (approved):
this looks great! thank you @vgwizardx !

markphelps commented:
> @markphelps I fixed the issues you mentioned. Since the srcWrong didn't show in Visual Studio, I forgot I removed it, but it must not have been deleted from the folder. Now, the header will be set by the strategy. As far as patterns for testing. Xunit is commonly used, similar to junit.
> 
> This might be more than what you want, but you can do something like this for the integration test https://blog.devgenius.io/integration-testing-in-net-8-b8ab83e7531d
> 
> You can also use other integration testing tools like Selenium, but I haven't set anything up like that.
> 
> If you just need something similar to how the Java test is set up. I probably could knock that out tonight.

No pressure! But if you are able to that would be amazing! yeah the tests are very basic at the moment, the [Java](https://github.com/flipt-io/flipt-server-sdks/blob/main/flipt-java/src/test/java/TestFliptClient.java) one is probably a good reference as you mentioned

basically it just depends on `FLIPT_URL` and `FLIPT_AUTH_TOKEN`.

If you are able to get something going I can take it and hook it up to our CI, as we use [Dagger](https://dagger.io/) and getting it all wired up takes a bit of time.

But if not I can also take a stab at writing a test if you don't have the time. Regardless this contribution is much appreciated! We've been asked by several people to have a C# SDK and now we can say we've got one!! Thanks again @vgwizardx
